packages/core/src/clients/ILeaderboardClient.ts

```diff
@@ -0,0 +1,51 @@
+import {
+  LeaderboardDelete,
+  LeaderboardFetch,
+  LeaderboardGetRank,
+  LeaderboardLength,
+  LeaderboardRemoveElements,
+  LeaderboardUpsert,
+} from '../messages/responses/leaderboard';
+import {
+  SortedSetFetchByRankCallOptions,
+  SortedSetFetchByScoreCallOptions,
+} from '../utils';
+
+export type LeaderboardFetchByRankOptions = SortedSetFetchByRankCallOptions;
+export type LeaderboardFetchByScoreOptions = SortedSetFetchByScoreCallOptions;
+
+export interface ILeaderboardClient {
+  leaderboardUpsert(
+    cacheName: string,
+    leaderboardName: string,
+    elements: Map<bigint, number>
+  ): Promise<LeaderboardUpsert.Response>;
+  leaderboardFetchByScore(
+    cacheName: string,
+    leaderboardName: string,
+    options?: LeaderboardFetchByScoreOptions
+  ): Promise<LeaderboardFetch.Response>;
+  leaderboardFetchByRank(
+    cacheName: string,
+    leaderboardName: string,
+    options?: LeaderboardFetchByRankOptions
+  ): Promise<LeaderboardFetch.Response>;
+  leaderboardGetRank(
+    cacheName: string,
+    leaderboardName: string,
+    elementId: bigint
```

Comment:
i started a thread on this in slack but we might want to make sure everyone is aligned on the name for this field. I think we might end up just shortening this to `id`, pending conversation.

Also might want to support `|number` here (unless TypeScript is kind enough to automatically cast a `number` to a bigint when you call a function, in which case you can ignore all similar feedback that I have left).